src/priorityq.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding utf-8 -*-
+
+
+class Priorityq(object):
+
+    def __init__(self, iterable=None):
+        '''Initialise a Priorityq from a list of lists or tuples
+            where the zeroeth element is the priority'''
+        if iterable is None:
+            self.priorityq = None
+        else:
+            try:
+                self.priorityq = sorted(iterable, key=lambda tup: tup[0])
+            except TypeError:
+                raise TypeError('please enter a list/tuple of '
+                                'lists/tuples with priority as '
+                                'the first value')
+
+    def insert(self, item):
+        ''' inserts an item into the queue.'''
+        if item is None:
+            raise ValueError('insert requires a tuple')
+        elif self.priorityq is None:
+            self.priorityq = [item]
+        else:
+            self.priorityq.append(item)
+            self.priorityq = sorted(self.priorityq, key=lambda tup: tup[0])
+
+    def pop(self):
+        '''removes the most important item from the queue.'''
+        if self.priorityq is None:
+            raise IndexError('tried to pop from an empty queue')
+        return self.priorityq.pop(0)
```

Comment:
You may not know this: `list.pop` at index 0 is a very expensive operation -- O(n) in fact, whereas `list.pop()` without specifying an index pops from the end of the list and is O(1). This is the sort of basic stuff you guys need to know -- review the big O notation reading and google search for time complexity guides for Python.